CONTRIBUTING.md

```diff
@@ -0,0 +1,107 @@
+
+> A properly formed Git commit subject line should always be able to complete the following sentence:
+> 
+> - If applied, this commit will <ins>your subject line here</ins>
+
+A simple but effective convention to follow for commits is the “problem / solution” pattern. It looks like this:
+```
+<Subject>
+
+Problem: <Statement of problem>
+
+Solution: <Statement of solution>
+```
+
+As an example, here is a commit taken from the rancher/rancher repo:
+```
+commit b71ce2892eecb7c87a5212e3486f1de899a694aa
+Author: Dan Ramich <danold215@gmail.com>
+Date:   Tue Jun 19 11:56:52 2018 -0700
+
+    Add Validator for RoleTemplate
+
+    Problem:
+    Builtin RoleTemplates can be updated through the API
+
+    Solution:
+    Add a Validator to ensure the only field that can be changed on a
+    builtin RoleTemplate is 'locked'
+```
+
+### Reviewing, addressing feedback, and merging
+Generally, pull requests need two approvals from maintainers to be merged.
+
+When addressing review feedback, it is helpful to the reviewer if additional changes are made in new commits. This allows the reviewer to easily see the delta between what they previously reviewed and the changes you added to address their feedback.
+
+Once a PR has at least two approvals, it can be merged. Here’s how the merge should be handled:
+- If the PR is a single logical commit, the merger should use the “Rebase and merge” option. This keeps the git commit history very clean and simple and eliminates noise from "merge commits."
+- If the PR is more than one logical commit, the merger should use the “Create a merge commit” option.
+- If the PR consists of more than one commit because the author added commits to address feedback, the commits should be squashed into a single commit (or more than one logical commit, if it is a big feature that needs more commits). This can be achieved in one of two ways:
+  - The merger can use the “Squash and merge” option. If they do this, the merger is responsible for cleaning up the commit message according to the previously stated commit message guidance.
+  - The pull request author, after getting the requisite approvals, can reorganize the commits as they see fit (using, for example, git rebase -i) and re-push.
+
+Once the PR is merged, please change the status of the linked issue in [rancher project](https://github.com/rancher/rancher) to "Needs QA". In the following days, the QA team will run tests and process the PR to verify that no regressions were introduced. Please be alert on the issue as the QA team might ask questions related to the PR. 
+
+### Backports
+
+#### RKE code
+
+To understand how backports should be pulled through, RKE project releases and what is included in each release should be described. RKE in general maintains two tracks in parallel which are differentiated by the minor version. Each of these tracks are developped in two different github branches:
```

Comment:
`release/v1.3` is not being actively maintained but will have critical bugs backported to it and released again if it's necessary to do so. It's not EOL.

Reply:
Thanks for this information, I was not aware of it. When does a version become EOL?

Reply:
I'm not super clear on the answer to this, but Rancher uses rke under the hood so it's tied to Rancher and the eol for supported kubernetes versions. The release/v1.3 line corresponds with Rancher 2.6 clusters and according to the support matrix, https://www.suse.com/suse-rancher/support-matrix/all-supported-versions/rancher-v2-6-13/ the latest Rancher 2.6 only supports up to k8s 1.24 clusters.

We are far beyond that now with k8s 1.26 reaching active EOL (and the maintenance phase) on Feb 28, 2024 https://kubernetes.io/releases/. Customers using rke 1.3.x are most likely in a position to upgrade than ask for new features or enhancements on their current version of rke. That being said, we still need to support customers on old versions so they are still in maintenance.

cc @kinarashah do we have any rke versions that are post-maintenance EOL?

Reply:
My understanding was that once k8s upstream version reaches EOL, we don't do any support to our customers or maybe just best effort. This will change when we announce the LTS kubernetes product which I don't know when will it happen

Reply:
My understanding was that once k8s upstream version reaches EOL, we don't do any support to our customers or maybe just best effort. This will change when we announce the LTS kubernetes product which I don't know when will it happen